### PR TITLE
Feat/item is gift

### PR DESCRIPTION
**Pull request by filafb**
#### What problem is this solving?

This code brings the `isGift` field into the `Item` query type. It supports a development of a feature request by Whirlpool, an European T1 client. The feature is in this [PR](https://github.com/vtex-apps/order-items/pull/42).

#### How should this be manually tested?

This [workspace](https://giftcart--itwhirlpool.myvtex.com) has this branch linked. Using the GraphQL IDE, it's possible to query this order `9a0266a45a8e4bac81d7be64b0c75959` that has an gift on it.

#### Checklist/Reminders

- [ ] Updated `README.md`.
- [x] Updated `CHANGELOG.md`.
- [ ] Linked this PR to a Jira story (if applicable).
- [ ] Updated/created tests (important for bug fixes).
- [ ] Deleted the workspace after merging this PR (if applicable).

#### Type of changes

✔️ | Type of Change
---|---
_ | Bug fix 
✔️ | New feature 
_ | Breaking change 
_ | Technical improvements

### CODE REVIEW

**Review by wdsrocha — CHANGES_REQUESTED** [review]
Could you add the `isGift` field on [the typing](https://github.com/vtex-apps/checkout-graphql/blob/main/node/typings/global.d.ts#L452-L481) as well? Besides that, could you rebase to the `main` branch?

**Comment by filafb** [comment]
> Could you add the `isGift` field on [the typing](https://github.com/vtex-apps/checkout-graphql/blob/main/node/typings/global.d.ts#L452-L481) as well? Besides that, could you rebase to the `main` branch?

Done!

**Comment by wdsrocha** [comment]
LGTM, but I'm unable to test this workspace, as I can't load your orderForm. I tried running the query bellow in the [graphql ide](https://giftcart--itwhirlpool.myvtex.com/admin/graphql-ide) but the flag `__RUNTIME__.settings['vtex.store'].advancedSettings.enableOrderFormOptimization` being set to `false` blocks this "loading orderForm" behavior.

```graphql
{
  orderForm(orderFormId: "9a0266a45a8e4bac81d7be64b0c75959") {
    id
  }
}
```

Can you tell me if there is any specific reason why this store is using both the `store-graphql` *and* `checkout-graphql` at the same time? If only checkout components are being used, you can remove the `store-graphql` dependency. Besides reducing the overall complexity, we would be able to enable the `enableOrderFormOptimization` flag and properly load an existent orderForm.

**Comment by wdsrocha** [comment]
Meanwhile, do you have any other way to manually test this? Maybe by adding the specifics products needed to trigger the gift

**Comment by filafb** [comment]
> 
> Can you tell me if there is any specific reason why this store is using both the `store-graphql` _and_ `checkout-graphql` at the same time? If only checkout components are being used, you can remove the `store-graphql` dependency. Besides reducing the overall complexity, we would be able to enable the `enableOrderFormOptimization` flag and properly load an existent orderForm.

I don't know the reason. I am tagging here @miguel-carrera. He is the one responsible for the project and might have an answer.

**Comment by filafb** [comment]
> LGTM, but I'm unable to test this workspace, as I can't load your orderForm. I tried running the query bellow in the graphql ide but the flag __RUNTIME__.settings['vtex.store'].advancedSettings.enableOrderFormOptimization being set to false blocks this "loading orderForm" behavior.

That is interesting. I am able to run the query over here. (Right now, the store took down the stock so you won't be able to see any product, but I took a print screen earlier).

Screen Shot 2021-04-08 at 4 01 41 PM

**Comment by wdsrocha** [comment]
> That is interesting. I am able to run the query over here.

The "problem" was that I couldn't test in the real store, but to be honest, testing only on the GraphQL IDE looks good enough already.

By the way, we still haven't approved the PR because this whole solution might conflict with some other behavior that will be implemented soon, so I'm checking with the design team what will be the best approach. Tomorrow I'll be able to give you feedback.

**Review by wdsrocha — APPROVED** [review]
As stated in the Slack thread, we will go with this solution. LGTM!

File: graphql/types/Item.graphql [hunk 1]
```diff
@@ -45,6 +45,7 @@ type Item {
   uniqueId: String!
   unitMultiplier: Float
   priceTags: [PriceTag!]!
+  isGift: Boolean
```
Comment (lucasecdb) [inline graphql/types/Item.graphql:48]: the `isGift` property is required, so we can type it as non-nullable

```suggestion
  isGift: Boolean!
```

Comment (filafb) [reply graphql/types/Item.graphql]: Done!

**Review by wdsrocha — APPROVED** [review]

**Review by lucasecdb — APPROVED** [review]

**Pull request merged**